### PR TITLE
Docs/api documentation

### PR DESCRIPTION
**Pull request by imnotteixeira**
Adding automated API documentation coded in Markdown!

You can preview the docs at the "netlify check" below!

### CODE REVIEW

**Comment by imnotteixeira** [comment]
For the reviewer, remind me to remove the pull request trigger on the workflow (only execute on push)

File: .github/workflows/docs.dev.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        DOC_BASE_FOLDER: ./documentation
+        ZIP_BUILD: false
+
+    - name: Deploy Documentation
+      uses: jsmrcaga/action-netlify-deploy@v1.1.0
+      with:
+        NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_TOKEN }}
+        NETLIFY_SITE_ID: ${{ secrets.NETLIFY_SITE_ID_DEV }}
+        NETLIFY_DEPLOY_TO_PROD: true
+        build_directory: documentation/build
```
Comment (andrefmrocha) [inline .github/workflows/docs.dev.yml:34]: Newline at the end of the file

File: .github/workflows/docs.master.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        DOC_BASE_FOLDER: ./documentation
+        ZIP_BUILD: false
+
+    - name: Deploy Documentation
+      uses: jsmrcaga/action-netlify-deploy@v1.1.0
+      with:
+        NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_TOKEN }}
+        NETLIFY_SITE_ID: ${{ secrets.NETLIFY_SITE_ID_MASTER }}
+        NETLIFY_DEPLOY_TO_PROD: true
+        build_directory: documentation/build
```
Comment (andrefmrocha) [inline .github/workflows/docs.master.yml:34]: Missing newline

**Review by andrefmrocha — COMMENTED** [review]
CI workflow looks great, don't want to approve it since I'm not if the offer API docs are correct and it probably for someone with more knowledge of the codebase to review that part

**Comment by imnotteixeira** [comment]
> CI workflow looks great, don't want to approve it since I'm not if the offer API docs are correct and it probably for someone with more knowledge of the codebase to review that part

Yep, a lot of actual docs are still missing... Will convert to draft in the meantime since the workflow verifiably works

**Comment by BrunoRosendo** [comment]
Now that we have documentation, we need to make sure this is always up to date whenever we change the API. Maybe we should create a pull request template to remind the contributors.
It would be awesome to have a bot detecting API changes but I don't know how easy that would be.

**Comment by Naapperas** [comment]
> 

I thought of this idea but it would be quite difficult to distinguish the change that happened to the API (or maybe I'm over complicating things in my head)

**Comment by Naapperas** [comment]
Maybe we could have a sub-section depicting how to build the documentation, what do you think ?

**Comment by BrunoRosendo** [comment]
> Maybe we could have a sub-section depicting how to build the documentation, what do you think ?

Do you mean the `npm run build`?

**Comment by Naapperas** [comment]
> > Maybe we could have a sub-section depicting how to build the documentation, what do you think ?
> 
> Do you mean the `npm run docs`?

Yes: its just that looking at the documentation, one can know right away how and where to write it but the build part was missing. I think it would be better if that information was also available on the documentation itself instead of the developer having to know the specific command.

**Comment by BrunoRosendo** [comment]
> 

Section added

**Comment by BrunoRosendo** [comment]
@Naapperas I included that the build is only needed for local development. I think everything is set now

**Review by Naapperas — APPROVED** [review]

**Pull request merged**